torch/lowrank.py

```diff
@@ -0,0 +1,249 @@
+"""Implement various linear algebra algorithms for low rank matrices.
+"""
+
+__all__ = ['svd', 'pca']
+
+import torch
+from . import _linalg_utils as _utils 
+
+
+def get_approximate_basis(A, q, niter=2, M=None):
+    """Return tensor :math:`Q` with q orthonormal columns such that
+    :math:`Q Q^H A` approximates :math:`A`. If :math:`M` is specified,
+    then `Q` is such that :math:`Q Q^H (A - M)` approximates :math:`A
```

Comment:
nit: matrix Q approximates the range of A, and, given that it has orthogonal columns, QQ*A approximates A.

Reply:
Also, if M is specified, is it that QQ*(A-M) approximates A or A-M?

Reply:
Approximation is in the spectral norm, right?

Reply:
> Also, if M is specified, is it that QQ*(A-M) approximates A or A-M?

It's `A-M`.

> The approximation is in the spectral norm, right?

AFAIK, it's 2-norm.

Reply:
Ok, then there is a typo in the description. Approximates A-M, not A